### PR TITLE
fix: Prevent saving a rule with an empty objectID

### PR DESCRIPTION
**Pull request by aseure**
Not sure if I missed to add some tests or even if it's the appropriate way of doing it. All comments are most welcome :)

### CODE REVIEW

File: Algolia.Search.Test/AlgoliaClientTest.cs [hunk 1]
```diff
@@ -1135,6 +1135,24 @@ public void TestSaveAndGetRule()
...
+        public void TestTrySaveRuleWithEmptyObjectID()
+        {
+            ClearTest();
+
+            string ruleID = "";
+            JObject rule = generateRuleStub(ruleID);
+            try
+            {
+				_index.SaveRule(rule);
+                Assert.
```
Comment (spinach) [inline Algolia.Search.Test/AlgoliaClientTest.cs]: So it seems there is a typo here. The build on ci.appvoyer is giving 
```
AlgoliaClientTest.cs(1148,24): error CS1001: Identifier expected [C:\projects\algoliasearch-client-csharp\Algolia.Search.Test\Algolia.Search.Test.csproj]
AlgoliaClientTest.cs(1148,24): error CS1002: ; expected [C:\projects\algoliasearch-client-csharp\Algolia.Search.Test\Algolia.Search.Test.csproj]
```

Basically either remove the assert or put something valid

File: Algolia.Search/Index.cs [hunk 2]
```diff
@@ -1680,6 +1680,12 @@ public Task<JObject> SaveRuleAsync(JObject queryRule, RequestOptions requestOpti
 			}
 
 			string objectID = (string)queryRule["objectID"];
+
+			if (objectID == "")
```
Comment (spinach) [inline Algolia.Search/Index.cs]: A better way to do it in C# is `objectID.IsNullOrEmpty`. nicer to read and covers the case of null as well.

**Comment by spinach** [comment]
@aseure Looks good in general, just 2 comments to address 👍

File: Algolia.Search.Test/AlgoliaClientTest.cs (hunk 1 above)
Comment (aseure) [reply Algolia.Search.Test/AlgoliaClientTest.cs]: Sorry about that, Visual Studio for Mac hasn't saved the file and I've committed without looking at the diff.

I've changed it a bit to make it use `Assert.Throws` like in the other tests.

File: Algolia.Search/Index.cs [hunk 3]
```diff
@@ -1680,6 +1680,12 @@ public Task<JObject> SaveRuleAsync(JObject queryRule, RequestOptions requestOpti
 			}
 
 			string objectID = (string)queryRule["objectID"];
+
+			if (objectID.IsNullOrEmpty)
```
Comment (spinach) [inline Algolia.Search/Index.cs]: woops my bad, that's supposed to be a method, so just add `()` at the end of `IsNullOrEmpty`

That's why the build failed in CI

Comment (aseure) [reply Algolia.Search/Index.cs]: Sorry but the Visual Studio for Mac wasn't helpful on that. It's gonna be easier with a Windows box soon :)

**Pull request merged**